src/wallet/wallet.cpp

```diff
@@ -6936,3 +7337,137 @@ bool CMerkleTx::AcceptToMemoryPool(const CAmount &nAbsurdFee, CValidationState &
 
 bool CompSigmaHeight(const CSigmaEntry &a, const CSigmaEntry &b) { return a.nHeight < b.nHeight; }
 bool CompSigmaID(const CSigmaEntry &a, const CSigmaEntry &b) { return a.id < b.id; }
+
+namespace {
+//I am leaving this proc in place as it may be useful for further urgent cases.
+[[deprecated]]
+void SendNotificationTx(CWallet * const pwallet, bip47::CPaymentChannel const & pchannel, CWalletTx& wtxNew)
+{
+    CAmount curBalance = pwallet->GetBalance();
+
+    if (curBalance < bip47::NotificationTxValue)
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Insufficient funds");
+
+    if (pwallet->GetBroadcastTransactions() && !g_connman) {
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
+    }
+
+    CBitcoinAddress const notifAddr = pchannel.getTheirPcode().getNotificationAddress();
+
+    // Parse Zcoin address
+    CScript scriptPubKey = GetScriptForDestination(notifAddr.Get());
+
+    // Create and send the transaction
+    CReserveKey reservekey(pwallet);
+    CAmount nFeeRequired;
+    std::string strError;
+    vector<CRecipient> vecSend;
+    int nChangePosRet = -1;
+    vecSend.push_back({scriptPubKey, bip47::NotificationTxValue, false});
+    CScript opReturnScript = CScript() << OP_RETURN << std::vector<unsigned char>(80);
+    vecSend.push_back({opReturnScript, 0, false});
+
+    if (!pwallet->CreateTransaction(vecSend, wtxNew, reservekey, nFeeRequired, nChangePosRet, strError))
+        throw JSONRPCError(RPC_WALLET_ERROR, strError);
+
+    if (wtxNew.tx->vin.size() == 0)
+        throw JSONRPCError(RPC_WALLET_ERROR, "Cannot select inputs for the notification tx");
+
+    CCoinsViewCache view(pcoinsTip);
+    const Coin coin = view.AccessCoin(wtxNew.tx->vin[0].prevout);
+    const CTxOut &prevout = coin.out;
+
+    CTxDestination dest;
+    CKey prevoutKey;
+    CKeyID keyID;
+    if (!ExtractDestination(prevout.scriptPubKey, dest) || !CBitcoinAddress(dest).GetKeyID(keyID) || !pwallet->GetKey(keyID , prevoutKey))
+        throw std::runtime_error("Cannot get the prevout key.");
+
+    bip47::Bytes const pcode = pchannel.getMaskedPayload(wtxNew.tx->vin[0].prevout, prevoutKey);
+    opReturnScript = CScript() << OP_RETURN << pcode;
+    vecSend[1].scriptPubKey = opReturnScript;
+
+    if (!pwallet->CreateTransaction(vecSend, wtxNew, reservekey, nFeeRequired, nChangePosRet, strError))
+        throw JSONRPCError(RPC_WALLET_ERROR, strError);
+
+    LogBip47("Sending pcode: %s to naddress: %s\n", pchannel.getMyPcode().toString(), notifAddr.ToString());
+
+    CValidationState state;
+    if (!pwallet->CommitTransaction(wtxNew, reservekey, g_connman.get(), state)) {
+        strError = strprintf("Error: The transaction was rejected! Reason given: %s", state.GetRejectReason());
+        throw JSONRPCError(RPC_WALLET_ERROR, strError);
+    }
+}
+
+void SendNotificationTxLelantus(CWallet * const pwallet, bip47::CPaymentChannel const & pchannel, CWalletTx& wtxNew)
```

Comment:
Definitely should be a method of `CWallet` class.

Reply:
I think it should not. If speak about `PrepareAndSendNotificationTx`, then maybe it should. But what for, what would be the benefits of this? It's a utility function as it just uses ::CWallet services to send a very special transaction.